FirebaseAuth/Sources/Utilities/FIRAuthRecaptchaVerifier.m

```diff
@@ -0,0 +1,256 @@
+                                           (void *)[RecaptchaClass methodForSelector:selector];
+                                       funcWithoutTimeout(
+                                           RecaptchaClass, selector, siteKey,
+                                           ^(id<RCARecaptchaClientProtocol> _Nullable recaptchaClient,
+                                             NSError *_Nullable error) {
+                                             if (recaptchaClient) {
+                                               self.recaptchaClient = recaptchaClient;
+                                               [self retrieveRecaptchaTokenWithAction:action
+                                                                           completion:completion];
+                                             } else if (error) {
+                                               completion(nil, error);
+                                             }
+                                           });
+                                     } else {
+                                       completion(nil,
+                                                  [FIRAuthErrorUtils recaptchaSDKNotLinkedError]);
+                                     }
+                                   } else {
+                                     completion(nil,
+                                                [FIRAuthErrorUtils recaptchaSDKNotLinkedError]);
+                                   }
+                                 } else {
+                                   NSLog(@"reCAPTCHA verification succeeded.");
+                                   [self retrieveRecaptchaTokenWithAction:action
+                                                               completion:completion];
+                                 }
+                               }];
+}
+
+- (void)retrieveRecaptchaConfigForceRefresh:(BOOL)forceRefresh
+                                 completion:(nullable FIRAuthRecaptchaConfigCallback)completion {
+  if (!forceRefresh) {
+    if (self.auth.tenantID == nil && _agentConfig != nil) {
+      completion(nil);
+      return;
+    }
+    if (self.auth.tenantID != nil && _tenantConfigs[self.auth.tenantID] != nil) {
+      completion(nil);
+      return;
+    }
+  }
+  FIRGetRecaptchaConfigRequest *request = [[FIRGetRecaptchaConfigRequest alloc]
+      initWithRequestConfiguration:self.auth.requestConfiguration];
+  [FIRAuthBackend
+      getRecaptchaConfig:request
+                callback:^(FIRGetRecaptchaConfigResponse *_Nullable response,
+                           NSError *_Nullable error) {
+                  if (error) {
+                    completion(error);
+                  }
+                  NSLog(@"reCAPTCHA config retrieval succeeded.");
+                  FIRAuthRecaptchaConfig *config = [[FIRAuthRecaptchaConfig alloc] init];
+                  config.siteKey = [response.recaptchaKey componentsSeparatedByString:@"/"][3];
```

Comment:
nit: add comment that sitekey is of the format - 'projects/<number>/keys/<siteKey>'